typescript/sdk/src/app/MultiProtocolApp.ts

```diff
@@ -120,6 +120,7 @@ export abstract class MultiProtocolApp<
     public readonly addresses: ChainMap<ContractAddrs>,
     public readonly logger = debug('hyperlane:MultiProtocolApp'),
   ) {
+    multiProvider = multiProvider.intersect(Object.keys(addresses)).result;
```

Comment:
I'm not strongly opposed but I don't think will match users intuition. Like say if they grab the MPP from an MPA later, I think they'd expect it to be the same one they initially gave. 

An alternative could be to add validation here to throw if the addresses map chains doesn't match the MPP chains. It would require us to intersect the MPP in Kathy here: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/rossy/cli-warp-deploy-artifact-selection/typescript/infra/scripts/helloworld/utils.ts#L131

If you prefer this approach, I'm okay with it